# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1247859623 14400
#      Fri Jul 17 15:40:23 2009 -0400
# Node ID bac81ac76a570d79ca68eb39b4f90da4b40fe3e4
# Parent  db08c23769708647f203198d3625358f9da33820
__go_draw_axes__.m: also use layer property for plot border

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2009-07-17  John W. Eaton  <jwe@octave.org>
+
+	* plot/__go_draw_axes__.m: Use "layer" property to decide whether
+	to send "set border front" or "set border layerdefault" to plot
+	stream.  Don't send "set border front" to plot stream for image data.
+
 2009-07-15  Robert T. Short <octave@phaselockedsystems.com>
 
 	* polynomial/conv.m: Fix Matlab incompatibility.  New tests.
 
 2009-07-09  John W. Eaton  <jwe@octave.org>
 
 	* plot/axis.m: Return 4-element vector for 2-d view.
 	From David Woodburn <david.woodburn@gmail.com>.
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -262,18 +262,23 @@ function __go_draw_axes__ (h, plot_strea
     else
       fputs (plot_stream, "set grid nomztics;\n");
     endif
 
     ## The grid front/back/layerdefault option also controls the
     ## appearance of tics, so it is used even if the grid is absent.
     if (strcmpi (axis_obj.layer, "top"))
       fputs (plot_stream, "set grid front;\n");
+      fputs (plot_stream, "set border front;\n");
     else
       fputs (plot_stream, "set grid layerdefault;\n");
+      ## FIXME -- the gnuplot help says that "layerdefault" should work
+      ## for set border too, but it fails for me with gnuplot 4.2.5.  So
+      ## use "back" instead.
+      fputs (plot_stream, "set border back;\n");
     endif
     if (! have_grid)
       fputs (plot_stream, "unset grid;\n");
     endif
 
     do_tics (axis_obj, plot_stream, ymirror, mono, gnuplot_term);
 
     xlogscale = strcmpi (axis_obj.xscale, "log");
@@ -364,17 +369,16 @@ function __go_draw_axes__ (h, plot_strea
 
 	  if (use_gnuplot_for_images)
 
 	    if (ndims (img_data) == 3)
 	      truecolor = true;
 	    elseif (strcmpi (obj.cdatamapping, "direct"))
 	      cdatadirect = true;
 	    endif
-	    fputs (plot_stream, "set border front;\n");
 	    data_idx++;
 	    is_image_data(data_idx) = true;
 	    parametric(data_idx) = false;
 	    have_cdata(data_idx) = false;
 	    have_3d_patch(data_idx) = false;
 
 	    [y_dim, x_dim] = size (img_data(:,:,1));
 	    if (x_dim > 1)
